digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/TrafficSignCsvImporter.scala

```diff
@@ -234,34 +244,53 @@ class TrafficSignCsvImporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl:
     }
 
+    def datesValidator(startDate: String, endDate: String) = {
+      try {
+        val startDateFormat = DateParser.DatePropertyFormat.parseDateTime(startDate)
+        val endDateFormat = DateParser.DatePropertyFormat.parseDateTime(endDate)
+
+        val isDatesOk = endDateFormat.isAfter(startDateFormat) || endDateFormat.isEqual(startDateFormat)
+        val errorMsg = if (isDatesOk) Nil else List("The end date value is equal/previous to the start date value.")
+
+        (isDatesOk, errorMsg)
+      } catch {
+        case _: Throwable => (false, List("Invalid dates formats"))
+      }
+    }
 
     val optTrafficSignType = getPropertyValueOption(parsedRow, "trafficSignType").asInstanceOf[Option[String]]
 
+    val (trafficSign, isSignOk) = optTrafficSignType match {
+      case Some(signType) =>
+        val sign = TrafficSignType.applyNewLawCode(signType)
+
+        if (sign == TrafficSignType.Unknown)
+          (sign, false)
+        else
+          (sign, true)
+
+      case _ => (TrafficSignType.Unknown, false)
+    }
+
+    //Validate if we get a valida sign
+    if (!isSignOk)
+      return (List("Not a valid sign!"), Seq() )
+
     /* start date validations */
     val temporaryDevices = Seq(4,5)
     val optLifeCycle = getPropertyValueOption(parsedRow, "lifeCycle").asInstanceOf[Option[Int]].getOrElse(PointAssetState.Unknown.value)
-    val optStartDate = getPropertyValueOption(parsedRow, "startDate").asInstanceOf[Option[String]]
-    val optEndDate = getPropertyValueOption(parsedRow, "endDate").asInstanceOf[Option[String]]
-
-    val (isValidDate, datesErrorMsg) = optTrafficSignType match {
-      case Some(signType) if (!isBlank(optLifeCycle.toString) && temporaryDevices.contains(optLifeCycle)) || TrafficSignType.applyNewLawCode(signType) == RoadWorks =>
-        (optStartDate, optEndDate) match {
-          case (Some(startDate), Some(endDate)) =>
-            try {
-              val startDateFormat = DateParser.DatePropertyFormat.parseDateTime(startDate)
-              val endDateFormat = DateParser.DatePropertyFormat.parseDateTime(endDate)
-
-              val isDatesOk = endDateFormat.isAfter(startDateFormat) || endDateFormat.isEqual(startDateFormat)
-              val errorMsg = if (isDatesOk) Nil else List("The end date value is equal/previous to the start date value.")
-
-              (isDatesOk, errorMsg)
-            } catch {
-              case _: Throwable => (false, List("Invalid dates formats"))
-            }
-          case (_, _) => (false, List("Invalid dates"))
-        }
-      case _ => (true, Nil)
-    }
+    val optStartDate = getPropertyValueOption(parsedRow, "startDate").asInstanceOf[Option[String]].map(_.trim).filterNot(_.isEmpty)
+    val optEndDate = getPropertyValueOption(parsedRow, "endDate").asInstanceOf[Option[String]].map(_.trim).filterNot(_.isEmpty)
+
+    val (isValidDate, datesErrorMsg) =
+      (optStartDate, optEndDate) match {
+        case (Some(startDate), Some(endDate)) =>
```

Comment:
Please create some unit tests to contemplate all of these conditions.

Reply:
Tests created